client/src/main/java/io/confluent/kafka/schemaregistry/client/CachedSchemaRegistryClient.java

```diff
@@ -31,14 +34,19 @@
 
 public class CachedSchemaRegistryClient implements SchemaRegistryClient {
 
-  private final String baseUrl;
+  private UrlList baseUrls;
   private final int identityMapCapacity;
   private final Map<String, Map<Schema, Integer>> schemaCache;
   private final Map<Integer, Schema> idCache;
   private final Map<String, Map<Schema, Integer>> versionCache;
 
-  public CachedSchemaRegistryClient(String baseUrl, int identityMapCapacity) {
-    this.baseUrl = baseUrl;
+  public static CachedSchemaRegistryClient newInstance(String baseUrlConfig, int identityMapCapacity) {
+    List<String> baseUrls = Arrays.asList(baseUrlConfig.split(","));
```

Comment:
It's a little hidden, but there's a empty check in the UrlList constructor. But yeah, I'll check out the config.getList method.